klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+                    rule_counts_df = rule_counts_df.rename(columns={"index": "rule_name"})
+
+                    rule_counts_df["table_name"] = tc_df.loc[tc_df["run_id"] == run_id, "table_name"].iloc[0]
+
+                    for c in ANALYSIS_RULES:
+                        if c not in rule_counts_df.columns:
+                            rule_counts_df[c] = 0
+
+                    rule_counts_df[ANALYSIS_RULES] = rule_counts_df[ANALYSIS_RULES].astype(int)
+                    rule_counts_df = rule_counts_df[["table_name", "rule_name"] + ANALYSIS_RULES]
+                    results_df_list.append(rule_counts_df)
+                    tc_df.loc[tc_df["run_id"] == run_id, "run_status"] = "completed"
+                else:
+                    tc_df.loc[tc_df["run_id"] == run_id, "run_status"] = "no output"
+
+            except Exception as exc:
+                logging.error("%d generated an exception: %s" % (run_id, exc))
+                traceback.print_exc()
+                tc_df.loc[tc_df["run_id"] == run_id, "run_status"] = "exception"
+
+    if len(results_df_list) > 0:
+        results_df = pd.concat(results_df_list)
+    else:
+        results_df = pd.DataFrame()
+
+    return results_df, tc_df
+
+
+def parse_existing_rules(rule_deck_path, output_path, target_table=None):
+    """
+    This function collects the rule names from the existing drc rule decks.
+
+    Parameters
+    ----------
+    rule_deck_path : string or Path object
+        Path string to the DRC directory where all the DRC files are located.
+    output_path : string or Path
+        Path of the run location to store the output analysis file.
+    target_table : string Optional
+        Name of the table to be in testing
+
+    Returns
+    -------
+    pd.DataFrame
+        A pandas DataFrame with the rule and rule deck used.
+    """
+
+    if target_table is None:
+        drc_files = glob.glob(os.path.join(rule_deck_path, "rule_decks", "*.drc"))
+    else:
+        table_rule_file = os.path.join(rule_deck_path, "rule_decks", f"{target_table}.drc")
+        if not os.path.isfile(table_rule_file):
+            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), table_rule_file)
+
+        drc_files = [table_rule_file]
+
+    rules_data = list()
+
+    for runset in drc_files:
+        with open(runset, "r") as f:
+            for line in f:
+                if ".output" in line:
+                    line_list = line.split('"')
+                    rule_info = dict()
+                    rule_info["table_name"] = os.path.basename(runset).replace(".drc", "")
+                    rule_info["rule_name"] = line_list[1]
+                    rule_info["in_rule_deck"] = 1
+                    rules_data.append(rule_info)
+
+    df = pd.DataFrame(rules_data)
+    df.drop_duplicates(inplace=True)
+    df.to_csv(os.path.join(output_path, "rule_deck_rules.csv"), index=False)
+    return df
+
+
+def analyze_test_patterns_coverage(rules_df, tc_df, output_path):
```

Comment:
should that be a separate tool?

Reply:
Not sure what's the requirement here.

Reply:
I was curious if this could be reported as a different job, see https://github.com/gdsfactory/gdsfactory/pull/1207#issuecomment-1409546645 as an example of a CI with separate coverage report.

Reply:
will be fixed in #9